rust/src/nfs/nfs2_records.rs

```diff
@@ -96,3 +96,211 @@ pub fn parse_nfs2_attribs(i: &[u8]) -> IResult<&[u8], Nfs2Attributes> {
+            0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x10, 0x10, 0x85,
+            0x00, 0x00, 0xb2, 0x5d, 0x38, 0x47, 0x75, 0xea,
+            0x00, 0x0b, 0x71, 0xb0, 0x38, 0x47, 0x71, 0xc4,
+            0x00, 0x08, 0xb2, 0x90, 0x38, 0x47, 0x75, 0xea,
+            0x00, 0x09, 0x00, 0xb0,
+
+        // data_len: (11)
+            0x00, 0x00, 0x00, 0x0b,
+
+        // data_contents: ("the b file")
+            0x74, 0x68, 0x65, 0x20,
+            0x62, 0x20, 0x66, 0x69, 0x6c, 0x65, 0x0a,
+
+        // _data_padding: parsed as part of data_contents
+            0x00,
+        ];
+
+        // since NfsReplyRead { data } is likely partial, parse/extract it here?
+        // then assert against it
+        fn extract_data(input: &[u8]) -> &[u8]  {
+
+            // strip packet_bytes up to data_contents
+            let i: IResult<&[u8], &[u8]> = take(76_usize)(input);
+            match i {
+                Ok((remainder, _input)) => {
+                        remainder
+                }
+                Err(error) => { panic!("extracting data field failed: {:?}", error); }
+            }
+        }
+
+        let expected_data = extract_data(buf);
+
+        let result = parse_nfs2_reply_read(buf);
+        match result {
+            Ok((r, response)) => {
+                assert_eq!(r.len(), 0);
+                assert_eq!(response.status, 0);
+                assert_eq!(response.attr_follows, 1);
+                assert_eq!(response.attr_blob.len(), 68);
+                assert_eq!(response.count, response.data_len);
+                assert_eq!(response.eof, false);
+                assert_eq!(response.data_len, 11);
+
+                // different assertions for data
+                assert_eq!(response.data, "the b file\n\0".as_bytes());
+                assert_eq!(response.data, expected_data);
+                assert_eq!(response.data.len(), 12);
+            }
+            Err(error) => { panic!("Parsing failed: {:?}", error); }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_attributes() {
+
+        // packet_bytes
+        let buf: &[u8] = &[
+
+        // Type: Regular File (1)
+            0x00, 0x00, 0x00, 0x01,
+
+        // Attributes
+        // _blob1
+            0x00, 0x00, 0x81, 0xa4,
+            0x00, 0x00, 0x00, 0x01,
+            0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x01,
+
+        // Size: (0)
+            0x00, 0x00, 0x00, 0x00,
+
+        // _blob2
+            0x00, 0x00, 0x40, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0x00, 0x10, 0x10, 0x85,
+            0x00, 0x00, 0xa3, 0xe7,
+            0x38, 0x47, 0x75, 0xea,
+            0x00, 0x08, 0x64, 0x70,
+            0x38, 0x47, 0x75, 0xea,
+            0x00, 0x08, 0x64, 0x70,
+            0x38, 0x47, 0x75, 0xea,
+            0x00, 0x08, 0x64, 0x70,
+        ];
+
+        let result = parse_nfs2_attribs(buf);
```

Comment:
IMO, its perfectly acceptable to just use `.unwrap()` here if we just end up panic'ing in the error case.  The output is clear with a line number that gets you directly where you need to get.  What you have here is fine as well, its just a little more work.